### PR TITLE
Run Travis CI against both JDK 7 and 8.

### LINKED ISSUE #547: Run CI against both JDK 7 and 8
Travis has support for testing against multiple JDK versions: http://blog.travis-ci.com/support_for_multiple_jdks/

We should update our CI config to test against both JDK 7 and 8 -- the two versions we currently support.

### PR DESCRIPTION
**Pull request by davidzchen**
Also, upgrade to Travis's container-based infrastructure.

Fixes #547

### CODE REVIEW

**Comment by davidzchen** [comment]
Ping.

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,7 @@
 languages: java
+jdk:
+  - openjdk7
```
Comment (logiclord) [inline .travis.yml]: why not open openjdk8 ?

Comment (davidzchen) [inline .travis.yml]: Unfortunately, Travis only supports oraclejdk8 for now and has not added support for openjdk8 yet. Adding openjdk8 to this config results in an error.

**Comment by logiclord** [comment]
LGTM

File: .travis.yml (hunk 1 above)
Comment (davidzchen) [inline .travis.yml]: I have added a TODO comment to add openjdk8 once it becomes available.

**Pull request merged**